OverlayPlugin.Core/MemoryProcessors/Combatant/LineCombatant.cs

```diff
@@ -0,0 +1,411 @@
+                            }
+                            if (combatant.PosZ != oldCombatant.PosZ)
+                            {
+                                changed.Add(combatant.GetType().GetField("PosZ"));
+                            }
+                        }
+                    }
+                    else if (combatant.Heading != oldCombatant.Heading)
+                    {
+                        var diff = Math.Abs((combatant.Heading + Math.PI) - (oldCombatant.Heading + Math.PI));
+                        var checkDiff = inCombat ? CombatantChangeCriteria.InCombatDistanceHeading : CombatantChangeCriteria.OutOfCombatDistanceHeading;
+                        if (diff > checkDiff)
+                        {
+                            changed.Add(combatant.GetType().GetField("Heading"));
+                        }
+                    }
+                }
+                if (changed.Count > 0)
+                {
+                    combatantStateMap[combatant.ID] = new CombatantStateInfo()
+                    {
+                        lastUpdated = now,
+                        combatant = combatant,
+                    };
+                    WriteLine(
+                        CombatantMemoryChangeType.Change,
+                        combatant.ID,
+                        string.Join("", changed.Select((fi) => FormatFieldChange(fi, combatant))));
+                }
+            }
+
+            // Any combatants no longer in memory, signify that they were removed
+            foreach (var ID in combatantStateMap.Keys)
+            {
+                // If we're filtering, only consider removing those in the filters
+                if (filter.Length > 0 && !filter.Contains(ID))
+                {
+                    continue;
+                }
+                if (!combatants.Any((c) => c.ID == ID))
+                {
+                    combatantStateMap.TryRemove(ID, out var _);
+                    WriteLine(CombatantMemoryChangeType.Remove, ID, "");
+                }
+            }
+        }
+
+        private string FormatFieldChange(FieldInfo info, Combatant combatant, bool skipDefaultValues = false)
+        {
+            var value = info.GetValue(combatant);
+
+            if (value == null)
+            {
+                if (skipDefaultValues)
+                {
+                    return string.Empty;
+                }
+                return $"|{info.Name}|NULL";
+            }
+
+            if (skipDefaultValues && value.Equals(CombatantChangeCriteria.DefaultValues[info.FieldType]))
+            {
+                return string.Empty;
+            }
+
+            if (info.Name == "PCTargetID" || info.Name == "NPCTargetID" || info.Name == "BNpcNameID" || info.Name == "BNpcID" || info.Name == "TargetID" || info.Name == "OwnerID" || info.Name == "CastTargetID")
+            {
+                return $"|{info.Name}|0x{value:X}";
```

Comment:
I'm not super keen on this, as I feel like it's inconsistent with other log lines.  I think it'd make sense to have these values match the `03` line by not prepending `0x`.